HACKING.md

```diff
@@ -4,70 +4,84 @@ Hacking on `snapd` is fun and straightforward. The code is extensively unit
 (including which version of the Go compiler).
 
-### Setting up your build environment
+### Setting up build environment
 
-If your Go environment (e.g. `GOPATH`) is already configured, you should skip
-this step. The Go environment setup can be added to your shell login script
-(e.g. `~/.bashrc` for bash) for automatic setup (after a terminal restart).
+`snapd` is published and maintained as a [Go module](https://go.dev/doc/modules/managing-source). 
+A module is a collection of [Go packages](https://go.dev/ref/spec#Packages) 
+stored in a file tree with a [go.mod](go.mod) file at its root. 
+The [go.mod](go.mod) file defines the module’s _module path_, 
+which is also the import path used for the root directory, 
+and its _dependency requirements_. 
+All this allows maintaining `snapd` outside of the strict hierarchy of `GOPATH`, 
+that was mandatory before the release of [Go 1.13](https://pkg.go.dev/go@go1.13)
+(see [build @Go1.10](https://pkg.go.dev/go/build@go1.10)).  
+
+Even with Go modules mode, `GOPATH` is still used by the Go tool implicitly, 
+to store downloaded source code and compiled commands 
+(see [Go docs](https://pkg.go.dev/cmd/go#hdr-GOPATH_and_Modules)).
+
+> If `$GOPATH` is not set explicitly, in Unix-like systems it defaults to `$HOME/go`.
+
+One can easily find the location of the `GOPATH` file-tree on any enabled machine
+by running the following command:
 
 ```
-export GOPATH=${HOME}/work
-mkdir -p $GOPATH
-export PATH="$PATH:$GOPATH/bin"
+go env GOPATH
 ```
 
-### Further environment variable details
-
-When working with the source of Go programs, you should define a path within
-your home directory (or other workspace) which will be your `GOPATH`. `GOPATH`
-is similar to Java's `CLASSPATH` or Python's `~/.local`. `GOPATH` is
-documented [online](http://golang.org/pkg/go/build/) and inside the `go` tool
-itself.
+#### Backward compatibility
 
-    go help gopath
+Some tools we use still rely on `$GOPATH/bin` being listed in `$PATH` 
+(see [Running spread with QEMU](#running-spread-with-qemu)).
+Add `$GOPATH/bin` to your `PATH` in your shell login script 
+(e.g. `~/.bashrc` for bash):
 
-Various conventions exist for naming the location of your `GOPATH`, but it
-should exist, and be writable by you. For example:
+```sh
+export PATH=$PATH:$(go env GOPATH)/bin
+```
 
-    export GOPATH=${HOME}/work
-    mkdir $GOPATH
+If your `GOPATH` contains more than one path, you need to add `/bin`
+suffix to each path:
 
-will define and create `$HOME/work` as your local `GOPATH`. The `go` tool
-itself will create three subdirectories inside your `GOPATH` when required;
-`src`, `pkg` and `bin`, which hold the source of Go programs, compiled
-packages and compiled binaries, respectively.
+```sh
+export PATH=$PATH:$(go env GOPATH | sed -e "s|:|/bin:|g")/bin
+```
 
-Setting `GOPATH` correctly is critical when developing Go programs. Set and
-export it as part of your login script.
+#### Setting custom `GOPATH` (_Optional_)
```

Comment:
I would actually vote for keeping this the suggested way, since it makes everything easier. I think we could make this document a bit slimmer if we just recommended this way (advanced users will find their own path anyway).

Reply:
Look, here I tend to disagree, once again since the `GOPATH` thing is considered a deprecated mod of operation. 
Moreover, if there are older projects that some of our contributors might use, who need this variable configured - we make their life more complicated without an actual need in it. 

Moreover, if any I would have recommended removing this section, if we are willing to make this document slimmer. It will also simplify the process of installation. If I don't need to configure something, why configuring it.

Reply:
I think we should write the minimal set of instructions to get the contributor's environment working. I agree that if `GOPATH` is no longer necessary, then we should not mention it.